### PR TITLE
feat: style checkbox

### PR DESCRIPTION
**Pull request by ArthurTriis1**
## What's the purpose of this pull request?
This PR styles the CheckBox component

## How it works? 
Add a stylesheet for CheckBox and implement a style like the [Figma Prototype](https://www.figma.com/file/7gfHfIzqJAbLlmB7fs16gQ/Handoff-v3-(PLP)?node-id=1455%3A71627)

Screen Shot 2022-01-20 at 11 59 20

### CODE REVIEW

**Review by filipewl — APPROVED** [review]
Looks great!

**Review by israelswf — APPROVED** [review]
✨

File: src/components/ui/Checkbox/checkbox.scss [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+  align-items: center;
+  justify-content: center;
+  width: rem(20px);
+  height: rem(20px);
+  border: var(--border-width-0) solid var(--color-border-input);
+  border-radius: var(--border-radius-default);
+  outline: none;
+  cursor: pointer;
+  appearance: none;
+
```
Comment (renatamottam) [inline src/components/ui/Checkbox/checkbox.scss:15]:
```suggestion
transition: border .2s ease, background-color .2s ease, box-shadow .2s ease;

```

**Review by renatamottam — APPROVED** [review]
Amazing!

File: src/components/ui/Checkbox/checkbox.scss [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+  &::before {
+    position: absolute;
+    top: rem(3px);
+    width: rem(6px);
+    height: rem(10px);
+    border: rem(2px) solid var(--bg-neutral-lightest);
```
Comment (renatamottam) [inline src/components/ui/Checkbox/checkbox.scss:21]:
```suggestion
    border: var(--border-width-1) solid var(--bg-neutral-lightest);
```

File: src/components/ui/Checkbox/checkbox.scss [hunk 3]
```diff
@@ -0,0 +1,99 @@
...
+    background-color: var(--bg-neutral-lightest);
+    border: var(--border-width-0) solid var(--bg-secondary-dark);
+
+    &::before {
+      position: absolute;
+      top: rem(8px);
+      width: rem(10px);
+      height: 0;
+      border-width: 0;
+      border-bottom: rem(2px) solid var(--bg-secondary-dark);
```
Comment (renatamottam) [inline src/components/ui/Checkbox/checkbox.scss:63]:
```suggestion
      border-bottom: var(--border-width-1) solid var(--bg-secondary-dark);
```

File: src/components/ui/Checkbox/checkbox.scss [hunk 4]
```diff
@@ -0,0 +1,100 @@
...
+    }
+  }
+
+  &[data-store-checkbox-partial="true"] {
+    background-color: var(--bg-neutral-lightest);
+    border: var(--border-width-0) solid var(--bg-secondary-dark);
+
+    &::before {
+      position: absolute;
+      top: rem(8px);
```
Comment (hellofanny) [inline src/components/ui/Checkbox/checkbox.scss:60]: Maybe use `top: auto` here? 🤔

**Review by hellofanny — APPROVED** [review]
Well doneee Arthur!!! Looks wonderful 🤩 🤩 🤩

File: src/components/ui/Checkbox/checkbox.scss [hunk 5]
```diff
@@ -0,0 +1,104 @@
...
+      background-color: var(--bg-disabled);
+      border: var(--border-width-0) solid var(--color-border-input-disabled);
+
+      &::before {
+        border-color: var(--color-border-input-disabled);
+      }
+    }
+
+    + label {
+      color: var(--color-text-disabled);
```
Comment (hellofanny) [inline src/components/ui/Checkbox/checkbox.scss:101]: Niceee!

File: src/components/ui/Checkbox/checkbox.scss (hunk 4 above)
Comment (ArthurTriis1) [reply src/components/ui/Checkbox/checkbox.scss:60]: yeah!

**Pull request merged**